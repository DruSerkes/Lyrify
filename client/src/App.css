html,
body,
#root {
	height: 100%;
	min-width: 100vw !important;
}

/* body {
	width: 100% !important;
} */
.App {
	text-align: center;
	height: 100vh;
	width: 100vw;
}

.App-header {
	background-color: #282c34;
	min-height: 100vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	font-size: calc(10px + 2vmin);
	color: white;
}

.App-link {
	color: #61dafb;
}

.Home h1 {
	margin-bottom: 1.3rem;
}

.Home-Login {
	opacity: 0;
	animation: fadeIn 1.8s forwards;
	animation-delay: 500ms;
	color: white !important;
}

.Navbar-Logo {
	border-right: 0.1em solid black;
	padding: 0.6rem;
	margin-right: 0.2rem;
	text-decoration: none;
}

.Navbar-Btn {
	margin: 0 0.6rem 0 0 !important;
	height: 100% !important;
}

.active {
	color: white !important;
}

.Playing-Header h1 {
	margin-bottom: 1.8rem;
}

/* .Playing button {
	margin: 0.75rem 1rem;
} */

.Lyrics-Button {
	margin: 0.5rem 1rem 0 1rem;
}

.Search-Header h1 {
	margin-bottom: 1.8rem;
}

.SearchForm {
	margin-bottom: 1rem;
}

.Lyrics {
	width: 85%;
	margin: 1rem auto;
}

.Lyrics-Content {
	padding: 0.3rem;
}

.Lyrics-Album {
	border: none;
}

.Lyrics-Header {
	position: relative;
	margin: 0.7rem auto;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
}

.Lyrics-Header-Left {
	margin-bottom: 1rem;
}

.Lyrics-Header-Right {
	position: relative;
}

.Lyrics-Album-Img {
	max-width: 95%;
	border-radius: 5px;
	height: auto;
	margin: 0 auto;
}

.Favorite {
	color: inherit;
	text-decoration: none;
}

.Favorite:hover {
	color: #ff5722;
	text-decoration: underline;
}

@keyframes fadeIn {
	0% {
		opacity: 0;
	}
	100% {
		opacity: 1;
	}
}

@media only screen and (max-width: 400px) {
	h1 {
		font-size: 4.7rem !important;
	}

	h4 {
		font-size: 2rem !important;
	}

	h5 {
		font-size: 1.5rem !important;
	}

	.Lyrics {
		width: 100%;
	}

	.Lyrics-Album-Img {
		max-width: 95%;
		border-radius: 5px;
		height: auto;
		margin: 0 auto;
	}

	.Lyrics-Album button {
		margin: 0.5rem 1rem 0 1rem;
	}
}
