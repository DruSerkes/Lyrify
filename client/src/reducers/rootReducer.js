import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
