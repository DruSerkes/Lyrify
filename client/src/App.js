import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getUser } from './reducers/actions';
import './App.css';
import Routes from './Routes';
import Navbar from './Navbar';

function App() {
	const user = useSelector((state) => state.user.data);
	const dispatch = useDispatch();

	useEffect(
		() => {
			if (user === {} && localStorage.getItem('id')) {
				const id = localStorage.getItem('id');
				dispatch(getUser(id));
			}
		},
		[ dispatch, user ]
	);

	return (
		<div className="App">
			<Navbar />
			<Routes />
		</div>
	);
}

export default App;
