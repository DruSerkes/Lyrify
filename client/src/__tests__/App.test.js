import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<App />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<App />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('should show login button to not-logged in user', () => {
	const { getByText } = render(
		<Provider store={store}>
			<MemoryRouter>
				<App />
			</MemoryRouter>
		</Provider>
	);
	const btn = getByText('Login with Spotify');
	expect(btn).toBeInTheDocument();
});
