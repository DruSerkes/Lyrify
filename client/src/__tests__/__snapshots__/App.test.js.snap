// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          aria-current="page"
          class="MuiBox-root MuiBox-root-4 Navbar-Logo active"
          href="/"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="Lyrify"
              class="MuiAvatar-img"
              src="logo500.png"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             yrify
          </p>
        </a>
      </div>
    </header>
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-5 Home"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Lyrify
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorSecondary"
        >
          <em>
            lyrics for the songs you love
          </em>
        </p>
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-Login MuiButton-containedPrimary"
            href="https://lyrify-server.herokuapp.com/spotify/auth"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Login with Spotify
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
